src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp

```diff
@@ -642,7 +643,11 @@ class npc_death_knight_initiate : public CreatureScript
 enum DarkRiderOfAcherus
 {
     SAY_DARK_RIDER              = 0,
-    SPELL_DESPAWN_HORSE         = 51918
+    SPELL_DESPAWN_HORSE         = 52267,
+
+    EVENT_START_MOVE            = 0,
```

Comment:
Never use 0 as Event ID, 0 is used by ExecuteEvent() to mean "no events was executed".
This makes me doubt about the quality of the whole PR. By looking at the code it feels like the EVENT_START_MOVE event is executed over and over and over. Did you debug your code to see what it was doing (with breakpoints and other debugging tools) ?